### PR TITLE
Move FlutterLogo out of Material library

### LINKED ISSUE #154448: Move FlutterLogo out of the Material library
### Use case

FlutterLogo is in the Material library, but has no Material dependencies/opinions. So it makes sense to move it out of the Material library.

Also, [CupertinoContextMenu's examples](https://api.flutter.dev/flutter/cupertino/CupertinoContextMenu-class.html?_gl=1*11gn7bu*_ga*ODkwNDg2NzQxLjE3MTM5OTkwNDQ.*_ga_04YGWK0175*MTcyNTA0NTMzOS4xMzkuMS4xNzI1MDUwMTA3LjAuMC4w) imports the Material library because of FlutterLogo. This move can help remove that import.

### Proposal

Move FlutterLogo out of the Material library.

### PR DESCRIPTION
**Pull request by SohanRaidev**
- Moved flutter_logo.dart from material to widgets
- Updated exports in material.dart and widgets.dart
- Updated imports in CupertinoContextMenu examples

Fixes #154448

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/widgets/flutter_logo.dart
Comment (navaronbracke) [inline packages/flutter/lib/src/widgets/flutter_logo.dart:1]: To verify that the new import works, this needs a test. However, we already have a test for `FlutterLogo` at 

https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/flutter_logo_test.dart

Could you move `flutter_logo_test.dart` to `test/widgets/flutter_logo_test.dart` (so essentially also moving the test file)
and update the import in that file to point to the widgets library? That would be sufficient as a test for this.
(updating the import statement _is_ the new test)

Comment (SohanRaidev) [reply packages/flutter/lib/src/widgets/flutter_logo.dart:1]: Hi @navaronbracke , I’ve made the requested changes. I’ve moved flutter_logo_test.dart to the widgets directory and updated the import to point to the widgets library. Please review when you have time. Thank you!

Comment (piedcipher) [reply packages/flutter/lib/src/widgets/flutter_logo.dart:1]: Test file is still under material directory.

**Comment by navaronbracke** [comment]
This still has an analysis failure:

```
An error was detected when looking at import dependencies within the Flutter package:
One of the files in the widgets package imports that package recursively.
```

If you run `dev/bots/analyze.dart` from your checkout of the Flutter SDK you can probably reproduce the failure.

**Comment by SohanRaidev** [comment]
@piedcipher @navaronbracke 

Thanks for pointing that out! I'll fix it as soon as possible and update the PR.

**Comment by Piinks** [comment]
@SohanRaidev there are still some failing checks here, can you take a look?

**Comment by victorsanni** [comment]
Closing as #155864 fixes the accompanying issue.

**Pull request closed without merging**